### PR TITLE
pylockfile: 0.9.1 -> 0.10.2

### PR DESCRIPTION
**Pull request by mbakke**
Requires #9550.

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -7367,19 +7367,26 @@ let
...
     src = pkgs.fetchurl {
-      url = "http://pylockfile.googlecode.com/files/${name}.tar.gz";
-      sha1 = "1eebaee375641c9f29aeb21768f917dd2b985752";
+      sha256 = "0zi7amj3y55lp6339w217zksn1a0ssfvscmv059g2wvnyjqi6f95";
+      url = "https://github.com/openstack/pylockfile/archive/${version}.tar.gz";
     };
 
-    # error: invalid command 'test'
     doCheck = false;
+    preConfigure = "export OSLO_PACKAGE_VERSION=${version}";
```
Comment (lethalman) [inline pkgs/top-level/python-packages.nix]: FYI you can set `OSLO_PACKAGE_VERSION = version;` at derivation level.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -7367,19 +7367,26 @@ let
...
+    name = "lockfile-${version}";
+    version = "0.10.2";
     src = pkgs.fetchurl {
-      url = "http://pylockfile.googlecode.com/files/${name}.tar.gz";
-      sha1 = "1eebaee375641c9f29aeb21768f917dd2b985752";
+      sha256 = "0zi7amj3y55lp6339w217zksn1a0ssfvscmv059g2wvnyjqi6f95";
+      url = "https://github.com/openstack/pylockfile/archive/${version}.tar.gz";
     };
 
-    # error: invalid command 'test'
```
Comment (lethalman) [inline pkgs/top-level/python-packages.nix]: Why do you remove this comment?

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -7367,19 +7367,26 @@ let
...
     doCheck = false;
+    preConfigure = "export OSLO_PACKAGE_VERSION=${version}";
+    buildInputs = with self; [
+      pbr nose (sphinx.override {src = pkgs.fetchurl {
```
Comment (lethalman) [inline pkgs/top-level/python-packages.nix]: What about updating sphinx instead?

**Comment by matthiasbeyer** [comment]
Commit message does not match [the guidelines](http://hydra.nixos.org/build/26115043/download/1/nixpkgs/manual.html#idm140737318595088)

**Comment by mbakke** [comment]
To address the comments here: this is a work-in-progress, and I removed the test comment because I intended to enable `doCheck` after #9549. However I was met with a nasty circular multi-dependency and have not had time to fix it. Requires adding older versions of some of the packages.

**Comment by domenkozar** [comment]
@mbakke care to fix this one?

**Comment by domenkozar** [comment]
Thanks!

**Pull request merged**